packages/core/admin/admin/src/pages/Users/ListPage/ModalForm/index.js

```diff
@@ -0,0 +1,198 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import {
+  Stack,
+  Grid,
+  GridItem,
+  ModalLayout,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  Button,
+  Breadcrumbs,
+  Crumb,
+  Box,
+  H2,
+} from '@strapi/parts';
+import { Formik } from 'formik';
+import { Form, GenericInput, useNotification, useOverlayBlocker } from '@strapi/helper-plugin';
+import { useQueryClient, useMutation } from 'react-query';
+import formDataModel from 'ee_else_ce/pages/Users/ListPage/ModalForm/utils/formDataModel';
+import roleSettingsForm from 'ee_else_ce/pages/Users/ListPage/ModalForm/utils/roleSettingsForm';
+import MagicLink from 'ee_else_ce/pages/Users/ListPage/ModalForm/MagicLink';
+import { axiosInstance } from '../../../../core/utils';
+import SelectRoles from './SelectRoles';
+import layout from './utils/layout';
+import schema from './utils/schema';
+import stepper from './utils/stepper';
+
+const ModalForm = ({ queryName, onToggle }) => {
+  const [currentStep, setStep] = useState('create');
+  const [isSubmitting, setIsSubmiting] = useState(false);
+  const [registrationToken, setRegistrationToken] = useState(null);
+  const queryClient = useQueryClient();
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { lockApp, unlockApp } = useOverlayBlocker();
+  const postMutation = useMutation(body => axiosInstance.post('/admin/users', body), {
+    onSuccess: async ({ data }) => {
+      setRegistrationToken(data.data.registrationToken);
+      await queryClient.invalidateQueries(queryName);
+      goNext();
+      setIsSubmiting(false);
+    },
+    onError: err => {
+      console.error(err.response);
+
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error', defaultMessage: 'An error occured' },
+      });
+    },
+    onSettled: () => {
+      unlockApp();
+    },
+  });
```

Comment:
Maybe we should put that mutation in a custom hook `useUsersMutation()` to remove the noise on that file.